src/main/java/uk/gov/companieshouse/web/companydocuments/controller/CompanyDetailsController.java

```diff
@@ -27,15 +27,15 @@ public CompanyDetailsController(CompanyService companyService,
     public String getCompanyDetails(@PathVariable String companyNumber,
                                     Model model) {
 
+        model.addAttribute("backButton", NavigationUrls.COMPANY_LOOKUP_FORWARD_TO_COMPANY_DETAILS);
         model.addAttribute("companyDetail", companyService.getCompanyDetail(companyNumber));
 
         return getTemplateName();
     }
 
     @PostMapping
     public String submitCompanyDetails(@PathVariable String companyNumber) {
-        return UrlBasedViewResolver.REDIRECT_URL_PREFIX + chsUrl + "/company-documents/company/"
-                        + companyNumber + "/list-company-documents";
+        return UrlBasedViewResolver.REDIRECT_URL_PREFIX + chsUrl + NavigationUrls.LIST_COMPANY_DOCUMENTS.replace("{companyNumber}", companyNumber);
```

Comment:
Maybe use Spring's `UriTemplate` and `expand` the company number into it, as opposed to a standard regex replace? Protects against malformed uri's 👍

Reply:
Unfortunately, can't use those in a @RequestMapping! 🙄

Reply:
Do you mean that you can't use UriTemplate in the body of the method?

Reply:
No, in the @RequestMapping. Spoken with Brad and converting value from NavigationUrls.java into a UriTemplate and then expanding in the body of methods

Reply:
I wonder if this is the sort of thing that @BSaunders95 was talking about?
https://www.baeldung.com/spring-uricomponentsbuilder